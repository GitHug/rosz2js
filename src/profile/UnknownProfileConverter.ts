import AbstractProfileConverter from './AbstractProfileConverter';
import { UnknownProfile, BSCharacteristic, TypeName } from '../types';

class UnknownProfileConverter extends AbstractProfileConverter<UnknownProfile, BSCharacteristic> {
  constructor(unexpectedTypeName: string) {
    super({
      typeName: TypeName.UNKNOWN,
      name: '-',
      unexpectedTypeName
    });
  }

  protected getProperty(bsCharacteristic: BSCharacteristic): Partial<UnknownProfile> {
    return { [bsCharacteristic.$.name]: bsCharacteristic._ };
  }
}

export default UnknownProfileConverter;
