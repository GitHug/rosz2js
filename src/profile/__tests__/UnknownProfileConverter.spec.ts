import UnknownProfileConverter from '../UnknownProfileConverter';
import { BSCharacteristic, BSProfile, TypeName } from '../../Types';

describe('UnknownProfileConverter', () => {
  it('should be able to handle characteristics that does not have a converter', () => {
    const converter = new UnknownProfileConverter('Weird Stuff');

    const bsProfile = {
      $: {
        typeName: 'Weird Unknown Type',
        name: 'Weird things'
      },
      characteristics: [
        {
          characteristic: [
            {
              $: {
                name: 'Super Strange'
              },
              _: 'Off the charts'
            },
            {
              $: {
                name: 'Some other stuff'
              },
              _: 'Stuff'
            }
          ]
        }
      ]
    } as unknown;

    const profile = converter.convert(bsProfile as BSProfile<BSCharacteristic>);

    expect(profile).toEqual({
      typeName: TypeName.UNKNOWN,
      unexpectedTypeName: 'Weird Stuff',
      name: 'Weird things',
      'Super Strange': 'Off the charts',
      'Some other stuff': 'Stuff'
    });
    expect(profile).toMatchSnapshot();
  });

  it('should be able to handle profiles without characteristics', () => {
    const converter = new UnknownProfileConverter('Weird Stuff');

    const bsProfile = {
      $: {
        typeName: 'Weird Unknown Type',
        name: 'Weird things'
      },
      characteristics: []
    } as unknown;

    const profile = converter.convert(bsProfile as BSProfile<BSCharacteristic>);

    expect(profile).toEqual({
      typeName: TypeName.UNKNOWN,
      unexpectedTypeName: 'Weird Stuff',
      name: 'Weird things'
    });
    expect(profile).toMatchSnapshot();
  });
});
