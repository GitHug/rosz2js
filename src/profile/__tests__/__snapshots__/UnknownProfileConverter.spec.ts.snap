// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownProfileConverter should be able to handle characteristics that does not have a converter 1`] = `
Object {
  "Some other stuff": "Stuff",
  "Super Strange": "Off the charts",
  "name": "Weird things",
  "typeName": "Unknown",
  "unexpectedTypeName": "Weird Stuff",
}
`;

exports[`UnknownProfileConverter should be able to handle profiles without characteristics 1`] = `
Object {
  "name": "Weird things",
  "typeName": "Unknown",
  "unexpectedTypeName": "Weird Stuff",
}
`;
